packages/mattermost-redux/src/selectors/entities/threads.ts

```diff
@@ -51,7 +67,10 @@ export function getThreads(state: GlobalState): IDMappedObjects<UserThread> {
 }
 
 export function getThread(state: GlobalState, threadId: $ID<UserThread> | undefined): UserThread | null {
-    if (!threadId || !getThreadsInCurrentTeam(state)?.includes(threadId)) {
+    if (
+        !threadId ||
+        !(getThreadsInCurrentTeam(state)?.includes(threadId) || getUnreadThreadsInCurrentTeam(state)?.includes(threadId))
```

Comment:
I'm curious why we check that the thread should be viewable in the current context here. I would've expected this to always return the thread if it's loaded.

Reply:
yep me too. but I was reluctant to remove the check...

Reply:
Hey @koox00, @hmhealey. A side effect of this selector is used in Global Threads to determine if the URL should be cleared or not. 

e.g. if the given threadId belongs to another team—this returns null, and `useThreadRouting().clear()` gets called.

`getThreadOrSynthetic` is the opposite—give it a fallback post and it'll always return something.